imagebuildah/stage_executor.go

```diff
@@ -1851,14 +1869,20 @@ func (s *StageExecutor) pullCache(ctx context.Context, cacheKey string) (referen
 			ReportWriter:        nil,
 			PullPolicy:          define.PullIfNewer,
 		}
-		id, err := buildah.Pull(ctx, src.DockerReference().String(), options)
+		srcString := ""
+		if src.Transport().Name() == imagedocker.Transport.Name() {
+			srcString = src.DockerReference().String()
+		} else if src.Transport().Name() == imagedirectory.Transport.Name() {
+			srcString = "dir://" + src.StringWithinTransport()
```

Comment:
`//` is not a part of the `dir:` syntax (and, theory corner, in POSIX, paths starting with `"//"` have, _hypothetically_ special meaning, some UNIX systems IIRC accepted a `"//nfs.host.name/path"` at the kernel level).

This should just unconditionally use `transports.ImageName`; the called function uses `alltransports.ParseImageName` (before starting with heuristics), so this should use `transports.ImageName` to create a clearly unambiguous input.